lib/bundler/metrics.rb

```diff
@@ -0,0 +1,141 @@
+      rescue Errno::ENOENT
+      end
+      @metrics_hash["command"] = ARGV.first
+      ruby = Bundler::RubyVersion.system
+      @metrics_hash["host"] = ruby.host
+      @metrics_hash["ruby_version"] = ruby.versions_string(ruby.versions)
+      @metrics_hash["bundler_version"] = Bundler::VERSION
+      @metrics_hash["rubygems_version"] = Gem::VERSION
+      if ruby.engine != "ruby"
+        # engine_version raises on unknown engines
+        engine_version = begin
+                            ruby.engine_versions
+                          rescue RuntimeError
+                            "???"
+                          end
+        @metrics_hash["ruby_engine"] = "#{ruby.engine}/#{ruby.versions_string(engine_version)}"
+      end
+      options = Bundler.settings.all.join(",")
+      @metrics_hash["options"] = options unless options.empty?
+      @metrics_hash["ci"] = cis.join(",") if cis.any?
+      # add any user agent strings set in the config
+      extra_ua = Bundler.settings[:user_agent]
+      @metrics_hash["extra_ua"] = extra_ua if extra_ua
+      add_additional_metrics
+      write_to_file
+    end
+
+    # cant use it for now 
+    # def add_performance_metrics(time)
+    #   @metrics_hash["command_time_taken"] = time.round(2)
+    # end
+
+    def send_metrics
+      # dummy server for now
+      begin
+        uri = URI.parse("https://webhook.site/ee1e4493-c0f0-4e40-84fb-3a36d79d47fb")
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = true
+        request = Net::HTTP::Post.new(uri.request_uri)
+        request.set_form_data(read_from_file)
+        http.request(request)
+      rescue SocketError
+        "TCP connection failed"
+      end
+
+      # We've sent the metrics so we empty the file
+      # File::TRUNC is preferable since File.truncate doesn't work for all systems
+      open(@path, File::TRUNC) if File.exist?(@path)
+    end
+
+  private
+
+    def write_to_file
+      SharedHelpers.filesystem_access(@path) do |file|
+        FileUtils.mkdir_p(file.dirname) unless File.exist?(file)
+        require "psych"
+        File.open(file, "a") {|f| f.write(Psych.dump(@metrics_hash)) }
+      end
+    end
+
+    def read_from_file
+      valid_file = @path.exist? && !@path.size.zero?
+      return {} unless valid_file
+      list = Array.new
+      SharedHelpers.filesystem_access(@path, :read) do |file|
+        require "psych"
+        Psych.load_stream(file.read) {|doc| list << doc }
+      end
+      list
+    end
+
+    def cis
+      env_cis = {
+        "TRAVIS" => "travis",
+        "CIRCLECI" => "circle",
+        "SEMAPHORE" => "semaphore",
+        "JENKINS_URL" => "jenkins",
+        "BUILDBOX" => "buildbox",
+        "GO_SERVER_URL" => "go",
+        "SNAP_CI" => "snap",
+        "CI_NAME" => ENV["CI_NAME"],
+        "CI" => "ci",
+      }
+      env_cis.find_all {|env, _| ENV[env] }.map {|_, ci| ci }
+    end
+
+    def add_additional_metrics
```

Comment:
Let's break this up into separate methods like `record_install_info` and `record_exec_info`.